packages/kbn-dev-utils/src/proc_runner/proc.ts

```diff
@@ -0,0 +1,180 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import execa from 'execa';
+import { statSync } from 'fs';
+
+import * as Rx from 'rxjs';
+import { tap, share, take, mergeMap, map, ignoreElements } from 'rxjs/operators';
+import chalk from 'chalk';
+
+import treeKill from 'tree-kill';
+import { promisify } from 'util';
+const treeKillAsync = promisify(treeKill);
+
+import { ToolingLog } from '../tooling_log';
+import { observeLines } from './observe_lines';
+import { createCliError } from './errors';
+
+const SECOND = 1000;
+const STOP_TIMEOUT = 30 * SECOND;
+
+export interface ProcOptions {
+  cmd: string;
+  args: string[];
+  cwd: string;
+  env?: Record<string, string | undefined>;
+  stdin?: string;
+}
+
+async function withTimeout(
+  attempt: () => Promise<void>,
+  ms: number,
+  onTimeout: () => Promise<void>
+) {
+  const TIMEOUT = Symbol('timeout');
```

Comment:
Using a symbol so it can't be coerced?  Or just for fun :)

Reply:
Eh :) mostly for peace of mind, so I can safely assume it was thrown by my code.